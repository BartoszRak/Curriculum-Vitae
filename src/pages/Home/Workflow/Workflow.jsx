import React from 'react'
import { FormattedMessage } from 'react-intl'
import { Typography, Divider, Container } from '@material-ui/core'

import useStyle from './Workflow.style'

export function Workflow() {
  const classes = useStyle()
  return (
    <section className={classes.section}>
      <Container classes={{ root: classes.root }} id="workflow">
        <Typography className={classes.title} variant="h1">
          <FormattedMessage id="pages.home.workflow.title" />
          <span aria-hidden className={classes.emoji}>
            ðŸ™Œ
          </span>
        </Typography>
        <Divider className={classes.divider} />
        <Typography className={classes.subtitle} variant="h3">
          <FormattedMessage id="pages.home.workflow.tiles.project.title" />
        </Typography>
        <Typography variant="body1">
          <FormattedMessage id="pages.home.workflow.tiles.project.description" />
        </Typography>
        <Typography className={classes.subtitle} variant="h3">
          <FormattedMessage id="pages.home.workflow.tiles.git.title" />
        </Typography>
        <Typography variant="body1">
          <FormattedMessage
            id="pages.home.workflow.tiles.git.description"
            values={{
              methodology: (
                <a
                  href="https://pl.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  git-flow
                </a>
              ),
            }}
          />
        </Typography>
      </Container>
    </section>
  )
}

export default Workflow
