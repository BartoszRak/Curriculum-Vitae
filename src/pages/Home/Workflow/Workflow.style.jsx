import { makeStyles } from '@material-ui/core'

export default makeStyles(({ breakpoints, palette, spacing }) => ({
  divider: {
    margin: spacing(2, 0),
  },
  emoji: {
    fontSize: '2rem',
    [breakpoints.down('md')]: {
      fontSize: '1.5rem',
    },
  },
  image: {
    width: '50%',
    height: '50%',
  },
  imageHolder: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
  },
  root: {
    width: '100%',
  },
  section: {},
  subtitle: {
    borderLeft: `4px solid ${palette.primary.main}`,
    marginBottom: spacing(1),
    marginTop: spacing(3),
    paddingLeft: spacing(2),
    [breakpoints.down('md')]: {
      fontSize: '1.8rem',
    },
  },
  title: {
    textShadow: '0 0 5px rgba(0, 0, 0, 0.2)',
    [breakpoints.down('md')]: {
      fontSize: '2rem',
    },
  },
}))
