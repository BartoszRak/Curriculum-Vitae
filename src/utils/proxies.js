/* eslint-disable */
export const createPropertyProxy = (obj = {}) => new Proxy(obj, {
  get: (target, name) => target[name] || `${name}Mock`,
})

export const PropertyProxy = createPropertyProxy()
