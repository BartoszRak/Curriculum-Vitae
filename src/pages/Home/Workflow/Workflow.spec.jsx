import React from 'react'
import { cleanup } from '@testing-library/react'
import { shallow } from 'enzyme'

import { Workflow } from './Workflow'
import { PropertyProxy } from '~utils/proxies'

jest.unmock('@material-ui/core')

let props

describe('component: Workflow', () => {
  afterEach(cleanup)

  beforeEach(() => {
    props = {
      classes: PropertyProxy,
    }
  })
  describe('rendering', () => {
    test('render without crash', () => {
      const wrapper = shallow(<Workflow {...props} />)

      expect(wrapper).toBeTruthy()
    })

    test('match snapshot', () => {
      const wrapper = shallow(<Workflow {...props} />)

      expect(wrapper).toMatchSnapshot()
    })
  })
})
