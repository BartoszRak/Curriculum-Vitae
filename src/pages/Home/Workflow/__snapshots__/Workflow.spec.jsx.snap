// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`global: Workflow should render as expected 1`] = `
<div
  id="workflow"
>
  <TypographyMock
    variant="h1"
  >
    How my workflow looks like?
    <span
      aria-hidden={true}
    >
      ðŸ™Œ
    </span>
  </TypographyMock>
  <DividerMock />
  <TypographyMock
    variant="h3"
  >
    Project side
  </TypographyMock>
  <TypographyMock
    variant="body1"
  >
    In case of project workflow I worked mostly in Agile & Scrum and in customized mixes of those two methodologies. I am fun of organized but flexible workflows that gives a possibility to adapt easily to the situation.
  </TypographyMock>
  <TypographyMock
    variant="h3"
  >
    Git part
  </TypographyMock>
  <TypographyMock
    variant="body1"
  >
    I always tend to use a workflow that is closest to
     
    <a
      href="https://pl.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow"
    >
      git-glow
    </a>
     
    methodology. When I got, a task I create one branch, in which I always try to create only one commit and I squash at the end of work - to be sure my main branch will not be cluttered.
  </TypographyMock>
</div>
`;
